### PR TITLE
MP and record damage ground effects

### PR DESCRIPTION
**Pull request by wlbragg**
These are the spark and smoke effects from collapsed gear, wing tips and tail drag.
Also refactored gearAGL.xml to ground-effects.xml. Created channel groundeffects in c172p-ground-effects.xml and ground-effects.xml. 
I have one more set to do and that is off road tire smoke.

### CODE REVIEW

File: Models/Effects/damage/left_gear.xml [hunk 1]
```diff
@@ -40,7 +47,13 @@
...
     </offsets>
 
-    <condition>
+	<condition>
+	    <greater-than>
+            <property alias="/params/left_gear_ground_effect_speed_kt/property"/>
+            <value>8</value>
+        </greater-than>
+    </condition>
+    <!--condition>
```
Comment (onox) [inline Models/Effects/damage/left_gear.xml]: Can you delete the commented code?

File: Models/Effects/damage/left_gear.xml [hunk 2]
```diff
@@ -28,8 +28,15 @@
 
 <PropertyList>
 
+  <params>
+    <!-- IMPORTANT: int properties 14 to 19, both included, are reserved for the immat system. Do not use them! -->
```
Comment (onox) [inline Models/Effects/damage/left_gear.xml]: It's not really necessary to put this warning here in all these effects files as long as it's put in `c172p-set.xml`. That's where you decide which multiplay property to use.

File: Systems/flight-recorder/components/effects.xml [hunk 3]
```diff
@@ -35,6 +35,36 @@
         <property type="string">/fdm/jsbsim/contact/unit[8]/WOW</property>
     </signal>
 
+	 <!-- Gear smoke ans spark-->
```
Comment (onox) [inline Systems/flight-recorder/components/effects.xml]: Small typo :smile:

**Comment by onox** [comment]
@wlbragg Can some of these `/fdm/jsbsim/contact/unit[i]/WOW` properties be removed from `Systems/flight-recorder/components/effects.xml`?

**Comment by onox** [comment]
I think `/fdm/jsbsim/contact/unit[6]/WOW`, `/fdm/jsbsim/contact/unit[7]/WOW`, and `/fdm/jsbsim/contact/unit[8]/WOW` can be removed. They are only used during non-replay mode.

**Comment by onox** [comment]
And `/fdm/jsbsim/contact/unit[3]/WOW`, `/fdm/jsbsim/contact/unit[4]/WOW`, and `/fdm/jsbsim/contact/unit[5]/WOW` as well.

**Comment by onox** [comment]
Good that you renamed `gearAGL.xml` to `ground-effects.xml` :+1:

**Comment by wlbragg** [comment]
Yes, I forgot to remove the comments, I'll remove them in my next commit over this branch with the off road smoke. I'll also clean up the recorder file as per your suggestion.

**Comment by onox** [comment]
Thanks.

**Comment by wlbragg** [comment]
OK, one question. Look at Effects/tyresmoke/tyre-smoke-????.xml

...xml
 <condition>
	  <or>
	    <property>/gear/gear[1]/tyre-smoke</property>
	    <greater-than>
            <property alias="/params/left_tyre_smoke_ground_effect_speed_kt/property"/>
            <value>100</value>
			
        </greater-than>
	  </or>
    </condition>
...
the property
...xml
  <property>/gear/gear[1]/tyre-smoke</property>
...
is only = 1 on a hard ground landing and only for a short time. It was for touchdown tire smoke.
I really wasn't sure how to handle it, I hated to write even more aliases for it.

**Comment by onox** [comment]
@wlbragg It was possible to use `/gear/gear[1]/tyre-smoke` in the JSBSim functions?

**Comment by onox** [comment]
Also you named everything `<left_tyre_smoke_ground_effect_speed_kt>` :smile:

**Comment by onox** [comment]
Btw, the old code says:

```xml
<greater-than>
    <property>fdm/jsbsim/ground/rolling_friction-factor</property>
    <value type="double">1</value>
</greater-than>
```

So, wouldn't it be better to replace `<property>/fdm/jsbsim/ground/rolling_friction-factor</property>` in `Systems/c172p-ground-effects.xml` with:

```xml
<gt>
    <property>/fdm/jsbsim/ground/rolling_friction-factor</property>
    <value>1</value>
</gt>
```

Or am I missing something?

**Comment by onox** [comment]
In that case you can just use the number 20 instead of 100 in the `Effects/tyresmoke/tyre-smoke-*.xml` files.

**Comment by onox** [comment]
About `<property>/gear/gear[1]/tyre-smoke</property>`. What you can do to get "OR" functionality, is use:

```xml
    <sum>
        <property>/gear/gear[0]/tyre-smoke</property>
        <product>
            ...
            <gt>
                <property>/velocities/groundspeed-kt</property>
                <value>20.0</value>
            </gt>
        </product>
    </sum>
</function>
<clipto>
    <max>1</max>
</clipto>
```

In the effect file you then have to remove the `<greater-than>` test because it has been moved to JSBSim.

**Comment by onox** [comment]
Actually, JSBSim has an `<or>` element: http://jsbsim.sourceforge.net/JSBSim/classJSBSim_1_1FGFunction.html

**Comment by wlbragg** [comment]
Thank onox, I reviewed you comments and will make changes this afternoon/evening.

**Comment by onox** [comment]
@wlbragg In case you have noticed that we've just merged some big branch to master, you don't need to merge master to this branch because Github says we can still merge this PR (+ it will not put an ugly "Merge master" commit in this feature branch :smile:).

**Comment by wlbragg** [comment]
"Also you named everything ```xml<left_tyre_smoke_ground_effect_speed_kt>```"
Where are you seeing this?

**Comment by wlbragg** [comment]
OK, I think I got it all. I can't believe I butchered it as bad as I did last night. I must have been really tired :(

**Comment by wlbragg** [comment]
I just got through testing this on MP and everything looks good. It is really cool to see all those effects carry over to MP.

**Comment by wlbragg** [comment]
Can we merge this?

**Comment by onox** [comment]
Let me check this.

**Comment by onox** [comment]
That last commit looks OK, so yes, :shipit:

**Comment by onox** [comment]
> "Also you named everything <left_tyre_smoke_ground_effect_speed_kt>"
Where are you seeing this?

I checked the diff again and it occurs in just one location, so I must have been very tired yesterday :tired_face:

**Comment by wlbragg** [comment]
What do you think about bringing the tire water spray over from Aircraft/Generic/Effects/ to local so we can add it to MP effects? Right now I think that is the only effect that isn't setup that way. I never did like the idea of centralized "common use" items. It might not be a bad thing if you could always guarantee they would hold all the advantage of a local object. But they hardly ever do for such little gain.

**Comment by onox** [comment]
That's true. In which files are those water spray effects referenced?

**Comment by onox** [comment]
I'm not completely sure which effect you're talking about right now.

**Comment by wlbragg** [comment]
Models/c172p.xml lines 909

**Comment by wlbragg** [comment]
It's just water spray off tires when raining.

**Comment by onox** [comment]
Ah, yeah, we should add that as well :smile:

**Comment by wlbragg** [comment]
Will do

**Comment by onox** [comment]
Btw, the maximum float and int properties can have index 19. So we have float[18] and float[19] available. Also, following properties shouldn't work:

```xml
274         <int n="20" alias="/fdm/jsbsim/wing-both/broken"/>
275         <int n="21" alias="/fdm/jsbsim/wing-damage/left-wing"/>
276         <int n="22" alias="/fdm/jsbsim/wing-damage/right-wing"/>
```

**Comment by onox** [comment]
To be honest, I'm not sure why we aren't using a `string` property like `string[0]` for the registration number. Now it's taking up precious properties :stuck_out_tongue_winking_eye:

**Comment by onox** [comment]
Although, we can (ab)use properties in `/engines`, or `/rotors` if we really want. See http://wiki.flightgear.org/Multiplayer_protocol They get transferred automatically, so you don't need to do any aliasing in `c172p-set.xml`.

**Comment by wlbragg** [comment]
Oops, 20-22 do work so what does that mean?

**Comment by onox** [comment]
They do work?

**Comment by onox** [comment]
They shouldn't because they aren't in the list.

**Comment by wlbragg** [comment]
I guess I'd better double check.

**Comment by wlbragg** [comment]
20 is not working

**Comment by wlbragg** [comment]
There not working. OK, time to refactor.

**Comment by onox** [comment]
Then 21 and 22 shouldn't be working either :grinning:

**Comment by wlbragg** [comment]
They aren't and 13 which is pontoon right broken (which is the next INT we use is working.

**Comment by onox** [comment]
That's beause 13 <= 19? :stuck_out_tongue_winking_eye:

**Comment by wlbragg** [comment]
10-4

**Comment by wlbragg** [comment]
Yes, just saying. I think I have some room to play with on some of the earlier damage effects, hopefully I can get enough freed up!

**Comment by onox** [comment]
Ok. Just know you can (ab)use the unused `/engines`, or `/rotors` properties if you want.

**Comment by wlbragg** [comment]
Looking at it again I think I am going to have to. I really thought I might be able to free some up but I don't see any.

**Comment by onox** [comment]
Can we try to move the immat properties to string[0] or string[1]?

**Comment by wlbragg** [comment]
Where are they?

**Comment by onox** [comment]
`/sim/multiplay/generic/string[0] `

**Comment by onox** [comment]
Does this registration number thing even work over MP?

**Comment by onox** [comment]
`Nasal/immat.nas` is only used locally.

**Comment by wlbragg** [comment]
No I don't see it over MP.

**Comment by wlbragg** [comment]
Unless it's used in pilot list or MP chat.

**Comment by onox** [comment]
You need to define it first in `c172p-set.xml`.

**Comment by wlbragg** [comment]
Which means they are free because it wasn't set up?

**Comment by onox** [comment]
Add this to `<multiplay><generic>`: `<string n="0" type="string">Hi there! :)</string>`.

**Comment by onox** [comment]
They are free yes, you can use them.

**Comment by onox** [comment]
Although, sometimes @andgi's MP broadcast system uses some of the string[] properties, so need to be careful with the index.

**Comment by wlbragg** [comment]
"@andgi's MP broadcast system uses some of the string[] properties"

What is that and which ones?

**Comment by onox** [comment]
I looked in my dual-control branch again, and it seems you need to specify which string[] property the MP broadcast system can use, so I think we can safely use string[0] for the registration number.

**Comment by wlbragg** [comment]
Oh, and then we can user 14-19?

**Comment by wlbragg** [comment]
They were using one INT per registration digit? How?

**Comment by onox** [comment]
The problem right now is that `Models/Immat/immat.xml` uses int[14] to int[19]. If we can find a way to just use string[0] then we can reclaim int[14] to int[19].

**Comment by wlbragg** [comment]
Right, looks like hey used each int to carry like a single digit ASCII char  code or something to that effect?

**Comment by onox** [comment]
Why is it using textranslate animation? Can't we just `<text>` or something?

**Comment by onox** [comment]
Like how the callsign was placed on the panel near the radio stack. (Old location)

**Comment by wlbragg** [comment]
This would require rewriting or replacing immat.nas

**Comment by onox** [comment]
That's not a problem for me. The `<animation>` stuff is (for me).

**Comment by onox** [comment]
If you are given a string property, are you able to place the text of it on the fuselage?

**Comment by wlbragg** [comment]
point me to the file and line#

**Comment by onox** [comment]
`Models/Immat/immat.xml` line 70 for the `<animation>`'s

**Comment by wlbragg** [comment]
Does canvas do text?

**Comment by onox** [comment]
Hmm... good idea. I think it can do text yes :smile:

**Comment by onox** [comment]
Shall we just temporarily nuke/freeze the immat system and reclaim int[14] to int[19]?

**Comment by wlbragg** [comment]
Works for me, we may get yelled at by the first person that uses it!

**Comment by onox** [comment]
Oh, I don't worry about that.

**Comment by onox** [comment]
Let's create a new issue for this.

**Comment by onox** [comment]
Too many comments in this PR already :stuck_out_tongue_winking_eye:

**Comment by wlbragg** [comment]
OK, I have to go, can you kill the immat and I will move the aliases over in the morning?

**Comment by onox** [comment]
Yes. See #299

**Comment by wlbragg** [comment]
OK, thanks, we'll go from there.

**Comment by onox** [comment]
Good idea from you to use Canvas :smile: :+1:

**Pull request merged**